File: trunk/ssi_30_mm/backend/theme_interface.php
<?php
require_once 'settings.php';

class ThemeInterface
{
	public $vars = NULL;
	public $customerId = NULL;
	public $themeName = NULL;
	public $homeURL = NULL;
	public $sessionId = NULL;
	public $config = NULL;
	
	public $HOME_URL = "";

	public function getVar($name)
	{
		if(isset($this->vars[$name])) return $this->vars[$name];
		if(!is_null($this->config) && isset($this->config->vars))
		{
			if(isset($this->config->vars->$name)) return $this->config->vars->$name; 
		}
		return '%' . $name . '%'; 
	}
	
	public function getVarHTML($name)
	{
		return htmlentities($this->getVar($name));   
	}
	
	public function getImageUrl($imageId, $noCaching = false)
	{
		return Settings::getImageUrl($imageId, $noCaching);
	}
	
	public function systemURL($url)
	{
		if(is_null($this->sessionId)) return $url;
		$url .= strpos($url, '?') ? '&' : '?';		
		$url .= 'sid=' . urlencode($this->sessionId);
		return $url; 
	}
}
?>

File: trunk/GenesysService/src/backend/Theme_Interface.java
package backend;

import java.net.URL;

public class Theme_Interface {
	class ThemeInterface {
		public String[] vars;
		public customerId = NULL;
		public themeName = NULL;
		public homeURL = NULL;
		public sessionId = NULL;
		public config = NULL;
	
		public String HOME_URL = "";

		public String getVar(String name)
		{
			if(vars[name] != null) return vars[name];
			if((config != null) && (config.vars != null))
			{
				if(config.vars.name != null) return config.vars.name; 
			}
			return "%" + name + "%"; 
		}
	
		public function getVarHTML(String name)
		{
			return htmlentities(this.getVar(name));   
		}
	
		public URL getImageUrl(String imageId, Boolean noCaching)
		{
			return Settings.getImageUrl(imageId, noCaching);
		}
		
		public URL getImageUrl(String imageId)
		{
			return Settings.getImageUrl(imageId, false);
		}
	
		public URL systemURL(URL url)
		{
			if(sessionId == null) return url;
			String temp = url.toString();
			temp.concat(temp.contains("?") ? "&" : "?");		
			url = temp.concat("sid=" + sessionId);
			return url; 
		}
	}
}
